GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -53,33 +51,24 @@ public RevisionDiffControl()
             _revisionDiffContextMenuController = new FileStatusListContextMenuController();
         }
 
-        public void ForceRefreshRevisions()
+        public void RefreshArtificial()
         {
-            var revisions = _revisionGrid.GetSelectedRevisions();
-            if (revisions.Count != 0)
+            if (!Visible)
             {
-                _oldRevision = revisions[0].Guid;
-                _oldDiffItem = DiffFiles.SelectedItem;
+                return;
             }
-            else
+
+            var revisions = _revisionGrid.GetSelectedRevisions();
+            if (!revisions.Any(r => r.IsArtificial))
             {
-                _oldRevision = null;
-                _oldDiffItem = null;
+                return;
             }
 
-            RefreshArtificial();
-        }
-
-        public void RefreshArtificial()
-        {
-            if (Visible)
+            DiffFiles.StoreNextIndexToSelect();
+            SetDiffs(revisions);
+            if (DiffFiles.SelectedItem == null)
             {
-                var revisions = _revisionGrid.GetSelectedRevisions();
-
-                if (revisions.Count > 0 && revisions[0].IsArtificial)
-                {
-                    DiffFiles.SetDiffs(revisions, _revisionGrid.GetRevision);
-                }
+                DiffFiles.SelectStoredNextIndex();
             }
```

Comment:
Call the identical `DisplayDiffTab` instead, please.

Reply:
Actually, in DisplayDiffTab(), SelectFirstVisibleItem() should be called instead, to select the first in the list if the currently selected cannot be "followed". Otherwise e.g. file with index 3 will be selected if that was selected before which can seem "random".
(Even if it generally is more useful to get to the same position in the list when switching commits).

Reply:
Yes, this will feel "random".
In addition, it happens that the file is found in the "Diff with BASE" only. This has confused me, especially if the heading is scrolled out of view.

Reply:
In general, it is a very useful feature when jumping between two commits.
There is a flicker yet: The FileDiff is cleared before the other diff of the same file is displayed.

Reply:
@mstv Are you OK with the limitations, can I squash and request a merge?

Reply:
This "random" selection happened a few times to me. I am afraid, it willl confuse users.

Reply:
You mean if the file is selected in BASE?
If so, the first file could be selected except for Artificial refresh.
(But I like to "follow" the file selected, even if not perfect.)

Reply:
> Otherwise e.g. file with index 3 will be selected if that was selected before which can seem "random".

Also this happened a few times today. Let me observe this another day.